src/plugins/ui_feature/find_and_replace.ts

```diff
@@ -114,6 +116,7 @@ export class FindAndReplacePlugin extends UIPlugin {
     if (this.isSearchDirty) {
       this.refreshSearch();
       this.isSearchDirty = false;
+      this.jumpToMatchSheet = true;
```

Comment:
I would add an argument `jumpToMatchSheet` to `refreshSearch` and `selectNextCell`, and set it to false here. Seems way less fishy than handle a plugin state